### PR TITLE
[Train] Remove deprecated mosaic and sklearn trainer code

### PR DESCRIPTION
**Pull request by Superskyyy**
https://github.com/ray-project/ray/issues/42257
https://github.com/ray-project/ray/pull/42814

## Why are these changes needed?
**Both trainers are now hard-deprecated.** This PR removes the code paths that confuses Ray train users on Mosaic and Sklearn. This should have been done a while ago.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Superskyyy** [comment]
I'm not sure about the mosaic util - RayLogger, is it still potentially required by TorchTrainer in any case?

File: python/ray/train/mosaic/__init__.py
Comment (matthewdeng) [inline python/ray/train/mosaic/__init__.py:1]: We can go ahead and remove this entirely, the `RayLogger` was only used as part of the `MosaicTrainer` before.

File: python/ray/train/mosaic/mosaic_trainer.py
Comment (matthewdeng) [inline python/ray/train/mosaic/mosaic_trainer.py:1]: Could you also do a quick scan to remove anything that might reference these? e.g. I saw there is a `python/ray/train/examples/mosaic_cifar10_example.py` that references the `MosaicTrainer`.

**Review by matthewdeng — COMMENTED** [review]
Thank you for doing this!

File: python/ray/train/mosaic/mosaic_trainer.py
Comment (Superskyyy) [reply python/ray/train/mosaic/mosaic_trainer.py:1]: Sure, I will do that

File: python/ray/train/mosaic/__init__.py
Comment (Superskyyy) [reply python/ray/train/mosaic/__init__.py:1]: Ok, thanks!

**Comment by anyscalesam** [comment]
@Superskyyy is this ready for re-review?

**Review by matthewdeng — APPROVED** [review]

**Pull request merged**